OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -290,25 +290,30 @@ public async Task<List<ShortEntityDto>> GetWorkshopListByEmployeeId(string emplo
     }
 
     /// <inheritdoc/>
-    public async Task<SearchResult<WorkshopProviderViewCard>> GetByProviderId(Guid id, ExcludeIdFilter filter)
+    public async Task<SearchResult<WorkshopProviderViewCard>> GetByProviderId(Guid id, WorkshopFilterTitle filter)
     {
         logger.LogInformation($"Getting Workshop by organization started. Looking ProviderId = {id}.");
 
-        filter ??= new ExcludeIdFilter();
-        ValidateExcludedIdFilter(filter);
+        filter ??= new WorkshopFilterTitle();
+        ValidateWorkshopTitleFilter(filter);
 
         var workshopBaseCardsCount = await workshopRepository.Count(whereExpression: x =>
-            filter.ExcludedId == null
-                ? (x.ProviderId == id)
-                : (x.ProviderId == id && x.Id != filter.ExcludedId)).ConfigureAwait(false);
+                (x.ProviderId == id) &&
+                (filter.ExcludedId == null || x.Id != filter.ExcludedId) &&
+                (string.IsNullOrEmpty(filter.SearchText) || x.Title.Contains(filter.SearchText)))
```

Comment:
Even if this expression correctly translates to SQL query - the resulting `WHERE` might be a bit too 'complex'.
You can use `PredicateBuilder` approach to apply different logic on dotnet side and have a very simple database predicate in the end.
Check private methods in the service and for example a `GetByFilter` method to see how it is used.